### PR TITLE
feat(Stoodi): new presence

### PR DESCRIPTION
**Pull request by DoginUwU**
## Description 
Created the presence for a Brazilian website focused on strengthening studies for the national high school exam

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<kbd></kbd>
<br />
<kbd></kbd>
<br />
<kbd></kbd>
<br />
<kbd></kbd>
</details>

### CODE REVIEW

File: websites/S/Stoodi/metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		"id": "890325980126789663",
+		"name": "Luiz ♡"
+	},
+	"service": "Stoodi",
+	"description": {
+		"en": "Stoodi is a platform to prepare you for exams and ENEM in Brazil.",
+		"pt_BR": "Stoodi é uma plataforma para te preparar para provas e ENEM no Brazil"
+	},
+	"url": "www.stoodi.com.br",
+	"regExp": "([a-z0-9]+)[.]stoodi.com.br",
```
Comment (theusaf) [inline websites/S/Stoodi/metadata.json:13]: Consistent regex, unless you meant to use the wildcard `.`
```suggestion
	"regExp": "([a-z0-9]+)[.]stoodi[.]com[.]br",
```

File: websites/S/Stoodi/metadata.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/wC4AkjI.png",
```
Comment (theusaf) [inline websites/S/Stoodi/metadata.json:15]: Logos must be [512x512px at minimum](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo).

File: websites/S/Stoodi/presence.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/wC4AkjI.png",
+			details: "Vendo o Stoodi",
+			startTimestamp: browsingTimestamp,
+		},
+		url = document.location.pathname;
+
+	if (url === "/") presenceData.details = "Navegando pela Home...";
```
Comment (theusaf) [inline websites/S/Stoodi/presence.ts:18]: Remove `...` from all strings

File: websites/S/Stoodi/presence.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		title = document.querySelector("h1.c-lesson__header-title")?.textContent;
+		if (subject && title) {
+			presenceData.details = `Vendo aula de de ${subject}:`;
+			presenceData.state = title;
+
+			const player: HTMLVideoElement =
+				document.querySelector(".fp-player video");
+
+			if (player) {
+				presenceData.smallImageKey = player.paused ? "pause" : "play";
```
Comment (theusaf) [inline websites/S/Stoodi/presence.ts:50]: Imgur links are preferred over Discord asset names.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/S/Stoodi/presence.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	strings = presence.getStrings({
+		play: "presence.playback.playing",
+		pause: "presence.playback.paused",
+	}),
+	images = {
+		play: "https://i.imgur.com/o4FaGfB.png",
+		pause: "https://i.imgur.com/kK39yND.png",
+	};
```
Comment (theusaf) [inline websites/S/Stoodi/presence.ts:12]: Prefer using `enum` for these instead.
Something like this:
https://github.com/PreMiD/Presences/blob/4f297578140efdf9ec96225629c892008df3839e/websites/C/Crunchyroll/presence.ts#L52-L58

File: websites/S/Stoodi/presence.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/wC4AkjI.png",
```
Comment (theusaf) [inline websites/S/Stoodi/presence.ts:16]: Also add the logo to the enum.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/S/Stoodi/presence.ts (hunk 5 above)
Comment (DoginUwU) [reply websites/S/Stoodi/presence.ts]: ohhh, okay

File: websites/S/Stoodi/presence.ts [hunk 7]
```diff
@@ -0,0 +1,131 @@
+const presence = new Presence({
+		clientId: "1043638037042712637",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000),
+	strings = presence.getStrings({
+		play: "presence.playback.playing",
+		pause: "presence.playback.paused",
+	});
+enum Assets {
+	play = "https://i.imgur.com/o4FaGfB.png",
```
Comment (DiscordAnaxes) [inline websites/S/Stoodi/presence.ts:10]: Enum's generally follow the Title case standard.

play -> Play
pause -> Pause

File: websites/S/Stoodi/presence.ts [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+				presenceData.state = title;
+			} else presenceData.details = "Procurando aulas ao vivo";
+		}
+		if (url.includes("/correcao-de-redacao")) {
+			title = document.querySelector(".c-page-header__title")?.textContent;
+			if (title) presenceData.details = "Vendo a correção de redação";
+			else presenceData.details = "Fazendo uma redação";
+		}
+		if (url.includes("/provas-do-enem")) {
+			title = document.querySelector(".c-page-header__title")?.textContent;
```
Comment (DiscordAnaxes) [inline websites/S/Stoodi/presence.ts:96]: Maybe create a global variable for this querySelector as it seems to be used quite extensively throughout the presence.

**Review by DiscordAnaxes — COMMENTED** [review]
If I recall correctly, presence.getStrings is used for multiLanguage presences. If this is the intention, it should be within a function called getStrings. Also, you should have a setting in metadata.json allowing multiLanguage. Have a look at Discord's presence, if you are stuck: https://github.com/PreMiD/Presences/blob/main/websites/D/Discord/presence.ts#L4

**Comment by DoginUwU** [comment]
> If I recall correctly, `presence.getStrings` is used for multiLanguage presences. If this is the intention, it should be within a function called `getStrings`. Also, you should have a setting in metadata.json allowing `multiLanguage: true`. Have a look at Discord's presence, if you are stuck: https://github.com/PreMiD/Presences/blob/main/websites/D/Discord/presence.ts#L4

Stoodi is only Brazilian, so I believe that it is not necessary the multilanguage function, just left as dynamic string the pause and play text, would still need this `multiLanguage: true`?

**Comment by DiscordAnaxes** [comment]
> > If I recall correctly, `presence.getStrings` is used for multiLanguage presences. If this is the intention, it should be within a function called `getStrings`. Also, you should have a setting in metadata.json allowing `multiLanguage: true`. Have a look at Discord's presence, if you are stuck: https://github.com/PreMiD/Presences/blob/main/websites/D/Discord/presence.ts#L4
> 
> Stoodi is only Brazilian, so I believe that it is not necessary the multilanguage function, just left as dynamic string the pause and play text, would still need this `multiLanguage: true`?

I'm not too sure honestly... Best to probably wait for Encrypted or theusaf to review the presence.

File: websites/S/Stoodi/presence.ts (hunk 6 above)
Comment (theusaf) [reply websites/S/Stoodi/presence.ts:17]: This shouldn't be resolved yet.

**Comment by theusaf** [comment]
> > If I recall correctly, `presence.getStrings` is used for multiLanguage presences. If this is the intention, it should be within a function called `getStrings`. Also, you should have a setting in metadata.json allowing `multiLanguage: true`. Have a look at Discord's presence, if you are stuck: https://github.com/PreMiD/Presences/blob/main/websites/D/Discord/presence.ts#L4
> 
> Stoodi is only Brazilian, so I believe that it is not necessary the multilanguage function, just left as dynamic string the pause and play text, would still need this `multiLanguage: true`?

In that case, you may not need to use `getStrings` at all, instead use the corresponding texts in place of them.

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**